textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -1,167 +1,222 @@
+
+            spanList = new ArrayList<>();
+            isPresent = false;
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
+    }
+
+    private Schema createSpanSchema() {
+        List<Attribute> dataTupleAttributes = schema.getAttributes();
+        Attribute[] spanAttributes = new Attribute[dataTupleAttributes.size() + 1];
+        for (int count = 0; count < spanAttributes.length - 1; count++) {
+            spanAttributes[count] = dataTupleAttributes.get(count);
+        }
+        spanAttributes[spanAttributes.length - 1] = SchemaConstants.SPAN_LIST_ATTRIBUTE;
+        Schema spanSchema = new Schema(spanAttributes);
+        return spanSchema;
+    }
+
+    /**
+     * @about Gets next matched tuple. Returns a new span tuple including the
+     *        span results. Performs a scan based search, gets the dictionary
+     *        value and scans all the documents for matches
+     * 
+     * @overview Loop through the dictionary entries. For each dictionary entry,
+     *           loop through the tuples in the operator. For each tuple, loop
+     *           through all the fields. For each field, loop through all the
+     *           matches.
+     */
+    @Override
+    public ITuple getNextTuple() throws Exception {
+        if (attributeIndex < searchInAttributes.size()) {
+            IField dataField = dataTuple.getField(searchInAttributes.get(attributeIndex).getFieldName());
+
+            if (dataField instanceof StringField || dataField instanceof TextField) {
+                String fieldValue = (String) dataField.getValue();
+
+                matcher = pattern.matcher(fieldValue.toLowerCase());
+                // Get position of dict value in the field.
+                if (matcher.find(positionIndex) != false) {
+
+                    isPresent = true;
+                    spanIndexValue = matcher.start();
+                    // Increment positionIndex so that next search occurs from
+                    // new positionIndex.
+                    positionIndex = spanIndexValue + dictionaryValue.length();
+
+                    // Attribute attribute =
+                    // schema.getAttributes().get(attributeIndex);
+                    spanFieldName = searchInAttributes.get(attributeIndex).getFieldName(); // attribute.getFieldName();
+
+                    addSpanToSpanList(spanFieldName, spanIndexValue, positionIndex, dictionaryValue, dictionaryValue);
+                    return getNextTuple();
+
+                } else {
+                    // Increment the fieldIndex and call getNextTuple to search
+                    // in next field
+                    attributeIndex++;
+                    positionIndex = 0;
+                    return getNextTuple();
+                }
+            } else {
+                // If fieldType is not String. Presently only supporting string

```

Comment:
"not String" -> "not String nor text"?

"supporting string" -> "supporting string and text"?